### PR TITLE
[Azure Search 2] Add DownloadSetComparer to detect download count changes

### PR DESCRIPTION
**Pull request by joelverhagen**
While comparing, fine places where download counts decrease and emit telemetry. This is a strange case currently causing some user pain and this will give us more data.

Progress on https://github.com/NuGet/NuGetGallery/issues/6458
Depends on https://github.com/NuGet/NuGet.Services.Metadata/pull/591
Will help us detect https://github.com/NuGet/NuGetGallery/issues/7258 is one more place

### CODE REVIEW

File: src/NuGet.Services.AzureSearch/Auxiliary2AzureSearch/DownloadSetComparer.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+        public SortedDictionary<string, long> Compare(
+            DownloadData oldData,
+            DownloadData newData)
+        {
+            var stopwatch = Stopwatch.StartNew();
+
+            // We use a very simplistic algorithm here. Find the union of both ID sets and compare each download count.
+            var uniqueIds = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            foreach (var id in oldData.Keys.Concat(newData.Keys))
```
Comment (loic-sharma) [inline src/NuGet.Services.AzureSearch/Auxiliary2AzureSearch/DownloadSetComparer.cs]: What're your thoughts on instead passing this LINQ `Concat` enumerable to the constructor of the `HashSet`?

Comment (loic-sharma) [reply src/NuGet.Services.AzureSearch/Auxiliary2AzureSearch/DownloadSetComparer.cs]: Same comment for line 83

File: tests/NuGet.Services.AzureSearch.Tests/Auxiliary2AzureSearch/DownloadSetComparerFacts.cs [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+                        true,
+                        7,
+                        true,
+                        false,
+                        0),
+                    Times.Once);
+            }
+
+            [Fact]
+            public void DetectsDecreaseDueToMissingId()
```
Comment (loic-sharma) [inline tests/NuGet.Services.AzureSearch.Tests/Auxiliary2AzureSearch/DownloadSetComparerFacts.cs]: Should you test the case where V1 increases by X, but V2 decreases by X, resulting in an empty delta but decrease telemetry? It's an weird/obscure corner case, feel free to ignore

File: src/NuGet.Services.AzureSearch/Auxiliary2AzureSearch/DownloadSetComparer.cs (hunk 1 above)
Comment (joelverhagen) [reply src/NuGet.Services.AzureSearch/Auxiliary2AzureSearch/DownloadSetComparer.cs]: Yeah, great idea! I forgot about that overload.

File: tests/NuGet.Services.AzureSearch.Tests/Auxiliary2AzureSearch/DownloadSetComparerFacts.cs (hunk 2 above)
Comment (joelverhagen) [reply tests/NuGet.Services.AzureSearch.Tests/Auxiliary2AzureSearch/DownloadSetComparerFacts.cs]: No this is a good thing to test. I'll add that.
